front/pages/w/[wId]/assistant/[cId]/index.tsx

```diff
@@ -39,68 +24,61 @@ export const getServerSideProps = withDefaultUserAuthRequirements<
   const subscription = auth.subscription();
 
   if (!owner || !user || !auth.isUser() || !subscription) {
+    const { cId } = context.query;
+
+    if (typeof cId === "string") {
+      return {
+        redirect: {
+          destination: `/w/${context.query.wId}/join?cId=${context.query.cId}`,
+          permanent: false,
+        },
+      };
+    }
+
     return {
       redirect: {
-        destination: `/w/${context.query.wId}/join?cId=${context.query.cId}`,
+        destination: "/",
         permanent: false,
       },
     };
   }
 
+  const { cId } = context.params;
+
   return {
     props: {
       user,
       owner,
       subscription,
       baseUrl: URL,
       gaTrackingId: GA_TRACKING_ID,
-      conversationId: context.params?.cId as string,
+      conversationId: typeof cId === "string" && cId !== "new" ? cId : null,
```

Comment:
`nit`: we are having this check a in 3 places what about we introduce a function?

Reply:
Added